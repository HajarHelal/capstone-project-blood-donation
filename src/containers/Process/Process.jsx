import React from 'react';
import { useTranslation } from 'react-i18next';
import Paragraph from '../../components/Paragraph/Paragraph';
import processImage from '../../components/images/centerImageProcess.png';
import Hero from '../../components/Hero/Hero';
import heroImage from '../../components/images/Man-Smiling-Donating-Blood.png';
import InstructionCard from '../../components/InstructionCard/InstructionCard';

function Process() {
  const { t } = useTranslation();
  const CardPosition = {
    cardStyle: ` mx-4 my-4`,
  };
  return (
    <div>
      <div>
        <Hero className="relative w-full h-full" img={heroImage}>
          <div className="absolute bottom-0 left-0 w-auto ml-6 mr-6 text-lg font-bold text-white sm:w-full md:w-3/4 lg:w-2/4 xl:w-2/4 sm:text-xl md:text-2xl">
            {t('Process.titlePrograph')}
          </div>
        </Hero>
      </div>

      <div className="mt-6">
        {' '}
        <Paragraph
          headerText={t('Process.titleInstructionCard')}
          paragraphText={t('Process.descTitleprog')}
          headerFlag
          paragraphFlag
        />
      </div>

      <div>
        <div className="grid grid-cols-1 mt-6 sm:grid-cols-1 md:grid-cols-2 lg:grid-cols-3 xl:grid-cols-3">
          <div>
            <div className={CardPosition.cardStyle}>
              <InstructionCard
                image="https://apifortress.com/doc/wp-content/uploads/2019/04/ForgeIcon_Black.svg"
                title={t('Process.age')}
                description={t('Process.descAge')}
                directionFlag="true"
              />
            </div>
            <div className={CardPosition.cardStyle}>
              <InstructionCard
                image="https://apifortress.com/doc/wp-content/uploads/2019/04/ForgeIcon_Black.svg"
                title={t('Process.weight')}
                description={t('Process.weightdesc')}
                directionFlag="true"
              />
            </div>
            <div className={CardPosition.cardStyle}>
              <InstructionCard
                image="https://apifortress.com/doc/wp-content/uploads/2019/04/ForgeIcon_Black.svg"
                title={t('Process.generalhealth')}
                description={t('Process.general_health_desc')}
                directionFlag="true"
              />
            </div>
          </div>

          <div className="hidden w-auto sm: md: lg:flex xl:flex ">
            <img src={processImage} alt="hel" />
          </div>

          <div>
            <div className={CardPosition.cardStyle}>
              <InstructionCard
                image="https://apifortress.com/doc/wp-content/uploads/2019/04/ForgeIcon_Black.svg"
                title={t('Process.eat')}
                description={t('Process.eat_desc')}
                // directionFlag="false"
              />
            </div>
            <div className={CardPosition.cardStyle}>
              <InstructionCard
                image="https://apifortress.com/doc/wp-content/uploads/2019/04/ForgeIcon_Black.svg"
                title={t('Process.drink')}
                description={t('Process.drinkdesc')}
                // directionFlag="false"
              />
            </div>
            <div className={CardPosition.cardStyle}>
              <InstructionCard
                image="https://apifortress.com/doc/wp-content/uploads/2019/04/ForgeIcon_Black.svg"
                title={t('Process.wait')}
                description={t('Process.waitdesc')}
                // directionFlag="false"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  );
}

export default Process;
