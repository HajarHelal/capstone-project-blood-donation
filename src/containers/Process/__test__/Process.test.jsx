import React from 'react';
import renderer from 'react-test-renderer';
import Process from '../Process';

// tests will use to compare to.
it('renders correctly', () => {
  const tree = renderer.create(<Process />).toJSON();
  expect(tree).toMatchSnapshot();
});
