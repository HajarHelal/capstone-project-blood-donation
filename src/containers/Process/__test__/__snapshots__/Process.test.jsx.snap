// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div>
    <div
      className="flex w-100  h-screen bg-no-repeat bg-cover"
      style={
        Object {
          "backgroundImage": "linear-gradient(rgba(0, 0, 0, 0.6),rgba(0, 0, 0, 0.6)),url(Man-Smiling-Donating-Blood.png)",
        }
      }
    >
      <div
        className="absolute bottom-0 left-0 w-auto ml-6 mr-6 text-lg font-bold text-white sm:w-full md:w-3/4 lg:w-2/4 xl:w-2/4 sm:text-xl md:text-2xl"
      >
        Process.titlePrograph
      </div>
    </div>
  </div>
  <div
    className="mt-6"
  >
     
    <div>
      <h1
        className="text-primary-100  sm:text-xl md:text-2xl lg:text-3xl pt-1 font-semibold  text-center"
      >
        Process.titleInstructionCard
      </h1>
      <p
        className=" pt-3 pb-5  sm:text-sm md:text-base lg:text-lg  text-center "
      >
        Process.descTitleprog
      </p>
    </div>
  </div>
  <div>
    <div
      className="grid grid-cols-1 mt-6 sm:grid-cols-1 md:grid-cols-2 lg:grid-cols-3 xl:grid-cols-3"
    >
      <div>
        <div
          className=" mx-4 my-4"
        >
          <div
            className="flex bg-white h-auto shadow-xl rounded w-full items-center p-4 md:p-6"
          >
            <div
              className="flex-none h-16 w-16 md:h-20 md:w-20 rounded-full m-auto mr-2 sm:mr-4  bg-gray-200"
            >
              <img
                alt="img"
                className="h-16 w-16 md:h-20 md:w-20 object-contain"
                src="https://apifortress.com/doc/wp-content/uploads/2019/04/ForgeIcon_Black.svg"
              />
            </div>
            <div
              className="flex-1 mr-4 sm:mr-4"
            >
              <p
                className="text-primary-100 text-lg sm:text-xl md:text-2xl font-bold"
              >
                Process.age
              </p>
              <p
                className="text-black text-sm sm:text-base md:text-lg tracking-wide leading-normal"
              >
                Process.descAge
              </p>
            </div>
          </div>
        </div>
        <div
          className=" mx-4 my-4"
        >
          <div
            className="flex bg-white h-auto shadow-xl rounded w-full items-center p-4 md:p-6"
          >
            <div
              className="flex-none h-16 w-16 md:h-20 md:w-20 rounded-full m-auto mr-2 sm:mr-4  bg-gray-200"
            >
              <img
                alt="img"
                className="h-16 w-16 md:h-20 md:w-20 object-contain"
                src="https://apifortress.com/doc/wp-content/uploads/2019/04/ForgeIcon_Black.svg"
              />
            </div>
            <div
              className="flex-1 mr-4 sm:mr-4"
            >
              <p
                className="text-primary-100 text-lg sm:text-xl md:text-2xl font-bold"
              >
                Process.weight
              </p>
              <p
                className="text-black text-sm sm:text-base md:text-lg tracking-wide leading-normal"
              >
                Process.weightdesc
              </p>
            </div>
          </div>
        </div>
        <div
          className=" mx-4 my-4"
        >
          <div
            className="flex bg-white h-auto shadow-xl rounded w-full items-center p-4 md:p-6"
          >
            <div
              className="flex-none h-16 w-16 md:h-20 md:w-20 rounded-full m-auto mr-2 sm:mr-4  bg-gray-200"
            >
              <img
                alt="img"
                className="h-16 w-16 md:h-20 md:w-20 object-contain"
                src="https://apifortress.com/doc/wp-content/uploads/2019/04/ForgeIcon_Black.svg"
              />
            </div>
            <div
              className="flex-1 mr-4 sm:mr-4"
            >
              <p
                className="text-primary-100 text-lg sm:text-xl md:text-2xl font-bold"
              >
                Process.generalhealth
              </p>
              <p
                className="text-black text-sm sm:text-base md:text-lg tracking-wide leading-normal"
              >
                Process.general_health_desc
              </p>
            </div>
          </div>
        </div>
      </div>
      <div
        className="hidden w-auto sm: md: lg:flex xl:flex "
      >
        <img
          alt="hel"
          src="centerImageProcess.png"
        />
      </div>
      <div>
        <div
          className=" mx-4 my-4"
        >
          <div
            className="flex sm:flex-row-reverse bg-white h-auto shadow-xl rounded w-full items-center p-4 md:p-6"
          >
            <div
              className="flex-none h-16 w-16 md:h-20 md:w-20 rounded-full m-auto mr-2 bg-gray-200"
            >
              <img
                alt="img"
                className="h-16 w-16 md:h-20 md:w-20 object-contain"
                src="https://apifortress.com/doc/wp-content/uploads/2019/04/ForgeIcon_Black.svg"
              />
            </div>
            <div
              className="flex-1 text-left sm:text-right mr-4 sm:mr-4"
            >
              <p
                className="text-primary-100 text-lg sm:text-xl md:text-2xl font-bold"
              >
                Process.eat
              </p>
              <p
                className="text-black text-sm sm:text-base md:text-lg tracking-wide leading-normal"
              >
                Process.eat_desc
              </p>
            </div>
          </div>
        </div>
        <div
          className=" mx-4 my-4"
        >
          <div
            className="flex sm:flex-row-reverse bg-white h-auto shadow-xl rounded w-full items-center p-4 md:p-6"
          >
            <div
              className="flex-none h-16 w-16 md:h-20 md:w-20 rounded-full m-auto mr-2 bg-gray-200"
            >
              <img
                alt="img"
                className="h-16 w-16 md:h-20 md:w-20 object-contain"
                src="https://apifortress.com/doc/wp-content/uploads/2019/04/ForgeIcon_Black.svg"
              />
            </div>
            <div
              className="flex-1 text-left sm:text-right mr-4 sm:mr-4"
            >
              <p
                className="text-primary-100 text-lg sm:text-xl md:text-2xl font-bold"
              >
                Process.drink
              </p>
              <p
                className="text-black text-sm sm:text-base md:text-lg tracking-wide leading-normal"
              >
                Process.drinkdesc
              </p>
            </div>
          </div>
        </div>
        <div
          className=" mx-4 my-4"
        >
          <div
            className="flex sm:flex-row-reverse bg-white h-auto shadow-xl rounded w-full items-center p-4 md:p-6"
          >
            <div
              className="flex-none h-16 w-16 md:h-20 md:w-20 rounded-full m-auto mr-2 bg-gray-200"
            >
              <img
                alt="img"
                className="h-16 w-16 md:h-20 md:w-20 object-contain"
                src="https://apifortress.com/doc/wp-content/uploads/2019/04/ForgeIcon_Black.svg"
              />
            </div>
            <div
              className="flex-1 text-left sm:text-right mr-4 sm:mr-4"
            >
              <p
                className="text-primary-100 text-lg sm:text-xl md:text-2xl font-bold"
              >
                Process.wait
              </p>
              <p
                className="text-black text-sm sm:text-base md:text-lg tracking-wide leading-normal"
              >
                Process.waitdesc
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="mx-6 my-6"
  >
    <h1
      className="pt-1 font-semibold text-center text-primary-100 sm:text-xl md:text-2xl lg:text-3xl"
    >
      Process.title_blood_donation
    </h1>
    <h1
      className="pt-1 font-semibold text-center text-black sm:text-xl md:text-2xl lg:text-3xll"
    >
      Process.secondtitle_blood_donation
    </h1>
    <p
      className="text-center sm:text-sm md:text-base lg:text-lg"
    >
       
      Process.blood_donation_desc
    </p>
  </div>
</div>
`;
